src/contracts/fungible/cache/file.rs

```diff
@@ -192,4 +192,178 @@ impl Cache for FileCache {
         self.save()?;
         Ok(existed)
     }
+
+    fn utxo_allocation_map(
+        &self,
+        contract_id: ContractId,
+    ) -> Result<HashMap<bitcoin::OutPoint, Vec<AtomicValue>>, CacheError> {
+        let asset = self.asset(contract_id).unwrap();
+
+        let allocation_map = asset.known_allocations();
+
+        let mut result = HashMap::new();
+
+        for item in allocation_map {
+            let mut alloc_amount = vec![];
+            for alloc in item.1 {
+                alloc_amount.push(alloc.value().value);
+            }
+            result.insert(item.0.clone(), alloc_amount);
+        }
```

Comment:
```suggestion
        self
            .asset(contract_id)?
            .known_allocations()
            .into_iter()
            .map(|outpoint, allocations| allocations.into_iter().map(|a| a.value().value).collect())
            .collect();
```
is the same without unwraps, clones, unnamed tuple parts, new vecs and hash map allocations.

Reply:
Fixed